go-controller/observability-lib/sampledecoder/sample_decoder.go

```diff
@@ -133,74 +135,60 @@ func (d *SampleDecoder) DecodeCookieIDs(obsDomainID, obsPointID uint32) (string,
-		case libovsdbops.BaselineAdminNetworkPolicyOwnerType:
-			msg = fmt.Sprintf("baseline admin network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.MulticastNamespaceOwnerType:
-			msg = fmt.Sprintf("multicast in namespace %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.MulticastClusterOwnerType:
-			msg = fmt.Sprintf("cluster multicast policy, direction %s", o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.NetpolNodeOwnerType:
-			msg = "default allow from local node policy, direction ingress"
-		case libovsdbops.NetworkPolicyOwnerType:
-			msg = fmt.Sprintf("network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.NetpolNamespaceOwnerType:
-			msg = fmt.Sprintf("network policies isolation in namespace %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.EgressFirewallOwnerType:
-			msg = fmt.Sprintf("egress firewall in namespace %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()])
-		case libovsdbops.UDNIsolationOwnerType:
-			msg = fmt.Sprintf("UDN isolation of type %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()])
-		}
-		return fmt.Sprintf("%s by %s", action, msg)
-	default:
-		return ""
+func newACLEvent(o *nbdb.ACL) *model.ACLEvent {
+	actor := o.ExternalIDs[libovsdbops.OwnerTypeKey.String()]
+	event := model.ACLEvent{
+		Action: o.Action,
+		Actor:  actor,
+	}
+	switch actor {
+	case libovsdbops.AdminNetworkPolicyOwnerType, libovsdbops.BaselineAdminNetworkPolicyOwnerType, libovsdbops.NetworkPolicyOwnerType:
+		event.Name = o.ExternalIDs[libovsdbops.ObjectNameKey.String()]
+		event.Direction = o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()]
+	case libovsdbops.MulticastNamespaceOwnerType, libovsdbops.NetpolNamespaceOwnerType:
+		event.Namespace = o.ExternalIDs[libovsdbops.ObjectNameKey.String()]
+		event.Direction = o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()]
+	case libovsdbops.MulticastClusterOwnerType:
+		event.Direction = o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()]
+	case libovsdbops.EgressFirewallOwnerType:
+		event.Namespace = o.ExternalIDs[libovsdbops.ObjectNameKey.String()]
+		event.Direction = string(libovsdbutil.ACLEgress)
+	case libovsdbops.UDNIsolationOwnerType:
+		event.Name = o.ExternalIDs[libovsdbops.ObjectNameKey.String()]
 	}
+	return &event
 }
 
-func (d *SampleDecoder) DecodeCookieBytes(cookie []byte) (string, error) {
+func (d *SampleDecoder) DecodeCookieBytes(cookie []byte) (model.NetworkEvent, error) {
```

Comment:
I'm purposely doing a breaking change here (since it's not GA feature yet, it's still time to break, right?)
@npinaeva can you point me to other consumers than netobserv where I should adapt code?